src/ERC7914.sol

```diff
@@ -14,6 +14,7 @@ abstract contract ERC7914 is IERC7914, BaseAuthorization {
     function approveNative(address spender, uint256 amount) external onlyThis returns (bool) {
         allowance[spender] = amount;
         emit ApproveNative(address(this), spender, amount);
+        emit AllowanceUpdated(spender, amount);
```

Comment:
do we really need this one and the `ApproveNative`? seems duplicative

Reply:
yeah wasn't sure - was thinking the same thing

Reply:
the allowance also gets updated on line 72, so thought maybe they should be the same event?

Reply:
I don't think we need both, im fine with keeping approve native

Reply:
Or, unless the spec defines that it must emit ApproveNative, could you just do AllowanceUpdated (since we use that in otehr places)

Reply:
oh hmm, maybe we can just not emit AllowanceUpdated here and assume that approvals will be indexed differently than allowances?

Reply:
okay yeah so approvals will then have to be indexed by both `ApproveNative` and `AllowanceUpdated`?